### PR TITLE
Check in/feature/adding emergency contact page pre check in

### PR DESCRIPTION
**Pull request by mdewey**
## Description
- Added support to show/hide emergency contact information confirmation page 

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/33297

## Testing done
- unit
- e2e

### CODE REVIEW

File: src/applications/check-in/pre-check-in/api/local-mock-api/mocks/v2/patient.pre.check.in.responses.js [hunk 1]
```diff
@@ -6,6 +6,23 @@ const createMockSuccessResponse = token => {
     id: token || defaultUUID,
     payload: {
       demographics: {
+        emergencyContact: {
+          name: 'Bugs Bunny',
```
Comment (acrollet) [inline src/applications/check-in/pre-check-in/api/local-mock-api/mocks/v2/patient.pre.check.in.responses.js:10]: 😆

File: src/applications/check-in/pre-check-in/utils/navigation/navigation.unit.spec.js [hunk 2]
```diff
@@ -28,15 +28,51 @@ describe('Pre-check in', () => {
...
-        const form = createForm({ hasConfirmedDemographics: false });
-        expect(form.length).to.equal(PRE_CHECK_IN_FORM_PAGES.length);
+        const form = createForm({
+          hasConfirmedDemographics: false,
+        });
+        // The '-1' is for not showing the emergency contact page
+        expect(form.length).to.equal(PRE_CHECK_IN_FORM_PAGES.length - 1);
       });
-      it('should not return the demographics or next of kin pages the pages when hasConfirmedDemographics is true', () => {
+      it('should not return the demographics,  next of kin, emergency contact pages the pages when hasConfirmedDemographics is true', () => {
```
Comment (acrollet) [inline src/applications/check-in/pre-check-in/utils/navigation/navigation.unit.spec.js:37]:
```suggestion
      it('should not return the demographics, next of kin, emergency contact pages when hasConfirmedDemographics is true', () => {
```

File: src/applications/check-in/pre-check-in/utils/navigation/navigation.unit.spec.js [hunk 3]
```diff
@@ -28,15 +28,51 @@ describe('Pre-check in', () => {
...
+      });
+      it('should return all pages with emergency contact page', () => {
+        const form = createForm({
+          hasConfirmedDemographics: false,
+          isEmergencyContactEnabled: true,
+        });
+        // The '-1' is for not showing the emergency contact page
+        expect(form.length).to.equal(PRE_CHECK_IN_FORM_PAGES.length);
+      });
+      it('should not return the demographics,  next of kin, emergency contact pages the pages when hasConfirmedDemographics is true and isEmergencyContact is true', () => {
```
Comment (acrollet) [inline src/applications/check-in/pre-check-in/utils/navigation/navigation.unit.spec.js:60]:
```suggestion
      it('should not return the demographics, next of kin, emergency contact pages when hasConfirmedDemographics is true and isEmergencyContact is true', () => {
```

File: src/applications/check-in/pre-check-in/pages/EmergencyContact/index.jsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  const selectCurrentContext = useMemo(makeSelectCurrentContext, []);
+  const { token } = useSelector(selectCurrentContext);
+
+  const selectForm = useMemo(makeSelectForm, []);
+  const { data } = useSelector(selectForm);
+  const { demographicsUpToDate, nextOfKinUpToDate } = data;
+
+  const selectVeteranData = useMemo(makeSelectVeteranData, []);
+  const { demographics } = useSelector(selectVeteranData);
+  const { emergencyContact } = demographics;
```
Comment (acrollet) [inline src/applications/check-in/pre-check-in/pages/EmergencyContact/index.jsx:39]: re-confirmed that this matches the key in LoROTA

**Review by acrollet — APPROVED** [review]
This should also get a review from another FE'er working on pre-checkin but I skimmed and re-confirmed the LoROTA key. 👍

**Review by brianseek — APPROVED** [review]
LGTM

**Pull request merged**